Add few ESLint errors (no-unused-vars)

If I try running ESLint (see config below), I get the following ESLint errors:

```sh
$ eslint .

/Users/pdehaan/dev/tmp/del/shield-studies-addon-utils/example/lib/variations.js
  10:5  error  'prefs' is defined but never used  no-unused-vars

/Users/pdehaan/dev/tmp/del/shield-studies-addon-utils/example/test/test-index.js
  1:5  error  'main' is defined but never used  no-unused-vars

/Users/pdehaan/dev/tmp/del/shield-studies-addon-utils/lib/index.js
    3:8  error  'Ci' is defined but never used  no-unused-vars
  104:2  error  Unnecessary semicolon           no-extra-semi

/Users/pdehaan/dev/tmp/del/shield-studies-addon-utils/test/test-xutils.js
  89:31  error  Parsing error: Unexpected token Boolean

✖ 5 problems (5 errors, 0 warnings)
```

### .eslintrc

```yaml
extends: eslint:recommended

env:
  es6: true
  node: true

rules:
  comma-dangle: 0
  no-console: 0
  no-unused-vars: [2, { vars: all, args: none }]
```

The line that ESLint really seems to be unable to parse is:
```js
const aConfig = xutils.xsetup(forSetup);
console.log(aConfig);
function hasVariationEffect() Boolean(prefSvc.get(FAKEPREF));
```

I think it doesn't like the lack of `{...}` around the function declaration and possibly the fact that we aren't returning a Boolean, just sort of converting something to a Boolean.

After converting that function to this:

```js
function hasVariationEffect() {
  return Boolean(prefSvc.get(FAKEPREF));
}
```

I get the following errors for that file:
```
/Users/pdehaan/dev/tmp/del/shield-studies-addon-utils/test/test-xutils.js
   39:4   error  Unnecessary semicolon             no-extra-semi
   54:9   error  'u' is defined but never used     no-unused-vars
  361:19  error  'seen' is defined but never used  no-unused-vars
```

--- a/.eslintignore
+++ b/.eslintignore
@@ -1,0 +1,2 @@
+coverage
+example

--- a/.eslintrc
+++ b/.eslintrc
@@ -1,0 +1,10 @@
+extends: eslint:recommended
+
+env:
+  node: true
+  es6: true
+
+rules:
+  comma-dangle: 0
+  no-console: 0
+  no-unused-vars: [2, { vars: all, args: none }]

--- a/Gruntfile.js
+++ b/Gruntfile.js
@@ -5,6 +5,9 @@ module.exports = function(grunt) {
     // gross, put this in the process
     process.env.coveragedir = require("os").tmpdir();
     grunt.initConfig({
+        eslint: {
+            files: '{,lib/,test/}*.js'
+        },
         shell: {
             jpmTest: {
                 command: 'jpm test',
@@ -37,6 +40,7 @@ module.exports = function(grunt) {
     });
 
 
+    grunt.loadNpmTasks('grunt-eslint');
     grunt.loadNpmTasks('grunt-babel');
     grunt.loadNpmTasks('grunt-shell');
     grunt.loadNpmTasks('grunt-istanbul');
@@ -48,9 +52,9 @@ module.exports = function(grunt) {
 
     if (process.env.TRAVIS) {
         grunt.log.ok("testing with travis path for fx");
-        grunt.registerTask('test', ['instrument', 'shell:jpmTestTravis', 'readcoverageglobal', 'storeCoverage', 'makeReport']);
+        grunt.registerTask('test', ['eslint', 'instrument', 'shell:jpmTestTravis', 'readcoverageglobal', 'storeCoverage', 'makeReport']);
     } else {
-        grunt.registerTask('test', ['instrument', 'shell:jpmTest', 'readcoverageglobal', 'storeCoverage', 'makeReport']);
+        grunt.registerTask('test', ['eslint', 'instrument', 'shell:jpmTest', 'readcoverageglobal', 'storeCoverage', 'makeReport']);
     }
 };
 

--- a/lib/index.js
+++ b/lib/index.js
@@ -1,6 +1,6 @@
 
 // Chrome privileged
-const {Ci, Cu} = require("chrome");
+const {Cu} = require("chrome");
 const { Services } = Cu.import("resource://gre/modules/Services.jsm");
 const { TelemetryController } = Cu.import("resource://gre/modules/TelemetryController.jsm");
 const CID = Cu.import("resource://gre/modules/ClientID.jsm");
@@ -101,7 +101,7 @@ function die (addonId=self.id) {
   if (prefSvc.get("shield.fakedie")) return;
   /* istanbul ignore next */
   require("sdk/addon/installer").uninstall(addonId);
-};
+}
 
 // TODO: GRL vulnerable to clock time issues #1
 function expired (xconfig, now = Date.now() ) {

--- a/package.json
+++ b/package.json
@@ -8,18 +8,29 @@
   },
   "author": "Gregg Lind <glind@mozilla.com>",
   "license": "ISC",
+  "dependencies": {
+  },
   "devDependencies": {
     "chai": "^3.5.0",
     "grunt": "^1.0.1",
     "grunt-babel": "^6.0.0",
     "grunt-cli": "^1.2.0",
+    "grunt-eslint": "^18.1.0",
     "grunt-istanbul": "^0.7.0",
     "grunt-shell": "^1.3.0",
-    "istanbul-jpm": "0.0.1",
+    "istanbul-jpm": "^0.0.1",
     "jpm": "^1.0.7"
   },
-  "bugs": "https://github.com/gregglind/shield-studies-addon-utils/issues",
+  "bugs": {
+    "url": "https://github.com/gregglind/shield-studies-addon-utils/issues"
+  },
   "files": [
     "lib/"
-  ]
+  ],
+  "repository": {
+    "type": "git",
+    "url": "git+https://github.com/gregglind/shield-studies-addon-utils.git"
+  },
+  "keywords": [],
+  "homepage": "https://github.com/gregglind/shield-studies-addon-utils#readme"
 }

--- a/test/test-xutils.js
+++ b/test/test-xutils.js
@@ -36,7 +36,7 @@ function only1Tab () {
       continue;
     }
     tab.close();
-  };
+  }
 }
 
 function hasTabWithUrlLike(aRegexp) {
@@ -51,7 +51,6 @@ function countTabsLike (aRegexp) {
   if (typeof aRegexp  === 'string') aRegexp = new RegExp(aRegexp)
   let n = 0;
   for (let tab of tabs) {
-    let u = tab.url;
     if (aRegexp.test(tab.url)) n += 1
   }
   return n;
@@ -86,7 +85,9 @@ const forSetup = {
 
 const aConfig = xutils.xsetup(forSetup);
 console.log(aConfig);
-function hasVariationEffect() Boolean(prefSvc.get(FAKEPREF));
+function hasVariationEffect() {
+  return Boolean(prefSvc.get(FAKEPREF));
+}
 
 /** Tests Begin Here */
 
@@ -356,7 +357,7 @@ exports['test 5: startup REVIVING a previous config keeps that config'] = functi
   expect(xconfig.variation).to.equal("e");
 
   // #3, do an install, and prove it did 'e'
-  let {thisStudy, seen, R} = setupStartupTest(xconfig, myVariations);
+  let {thisStudy, R} = setupStartupTest(xconfig, myVariations);
   promiseFinalizedStartup(thisStudy).then(waitABit).then(
   ()=>{
     expect(prefSvc.get(FAKEPREF)).to.be.equal("e");

